x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ExpressionBuilder.java

```diff
@@ -233,7 +233,7 @@ public Object visitQualifiedIntegerLiteral(EsqlBaseParser.QualifiedIntegerLitera
                 case "month", "months" -> Period.ofMonths(safeToInt(safeToLong(value)));
                 case "year", "years" -> Period.ofYears(safeToInt(safeToLong(value)));
 
-                default -> throw new ParsingException(source, "Unexpected numeric qualifier '{}'", qualifier);
+                default -> throw new ParsingException(source, "Unexpected time interval '{}'", qualifier);
```

Comment:
We only have time spans for qualifiers, but i think the vision was that we could add other non-time units as well. But not sure if this will happen and if it will, we can change this again. However, I think what's wrong is still the qualifier (vs., for instance, giving a time interval that was unexpectedly large/small):

```suggestion
                default -> throw new ParsingException(source, "Unexpected time interval qualifier: '{}'", qualifier);
```